File: solutions/elixir/resistor-color-duo/2/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @moduledoc false

  @codes %{
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9
  }

  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value([a, b | _]) do 
    @codes[a] * 10 + @codes[b]
  end
end

File: solutions/typescript/resistor-color-duo/2/resistor-color-duo.ts
type Color = (typeof COLORS)[number];

export function decodedValue(colors: Color[]): number {
  const [c1, c2] = colors;
  return COLORS.indexOf(c1) * 10 + COLORS.indexOf(c2);
}

export const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
] as const;
